File: tests/fixtures/dart/user_service.dart
import 'package:flutter/material.dart';
import 'package:dio/dio.dart';
import 'models.dart';

abstract class Repository {
  Future<User?> find(String id);
  Future<void> save(User user);
}

class HttpRepository implements Repository {
  final Dio client;

  HttpRepository(this.client);

  @override
  Future<User?> find(String id) async {
    final response = await client.get('/users/$id');
    return User.fromJson(response.data);
  }

  @override
  Future<void> save(User user) async {
    await client.post('/users', data: user.toJson());
  }
}

class UserService {
  final Repository repo;
  UserService(this.repo);

  Future<User?> find(String id) => repo.find(id);
}

mixin Loggable {
  void log(String msg) => print(msg);
}

int helper(int a, int b) {
  return a + b;
}

File: tests/fixtures/scala/UserService.scala
package com.example.svc

import scala.collection.mutable
import org.example.User

trait UserRepository {
  def find(id: String): Option[User]
  def save(user: User): User
}

case class User(id: String, name: String)

class UserService(repo: UserRepository) {
  def find(id: String): Option[User] = repo.find(id)

  def create(name: String): User = {
    val u = User(id = "x", name = name)
    repo.save(u)
  }
}

object Constants {
  val PAGE_SIZE = 20
  def helper(a: Int, b: Int): Int = a + b
}
